module/importer-magento-1/src/Infrastructure/Processor/StartMagento1ImportProcess.php

```diff
@@ -98,35 +96,50 @@ public function start(Import $import): void
 
         $result = [];
         $message = null;
+        $this->reader->open($import);
         try {
-            $products = $this->reader->read($source, $import, $transformer);
-            foreach ($products as $sku => $product) {
-                $result[$sku] = new ProductModel();
-                foreach ($product as $code => $version) {
-                    $result[$sku]->set($code, $version);
-                }
+            $product = $this->reader->read($transformer);
+
+            while ($product !== null) {
```

Comment:
```suggestion
            while ($product = $this->reader->read($transformer)) {
```
No need for duplication